### PR TITLE
java 9 migration

### PR DESCRIPTION
**Pull request by qweek**
add jdk 9 to travis
compile with java 9

#### Fix dependency error [stackoverflow/java-9-how-to-get-access-to-javax-annotation-resource-at-run-time](https://stackoverflow.com/questions/46502001/java-9-how-to-get-access-to-javax-annotation-resource-at-run-time)
java.xml.ws.annotation module is deprecated for removal
```
error: cannot find symbol
@javax.annotation.Generated(
                 ^
  symbol:   class Generated
  location: package javax.annotation
```

### CODE REVIEW

**Comment by qweek** [comment]
#### Known issue [netty/netty/#7254](https://github.com/netty/netty/issues/7254)
Illegal reflective access warnings when running Netty with Java 9
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by io.netty.util.internal.ReflectionUtil (file:netty-common-4.1.16.Final.jar) to constructor java.nio.DirectByteBuffer(long,int)
WARNING: Please consider reporting this to the maintainers of io.netty.util.internal.ReflectionUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

**Comment by qweek** [comment]
#### Possible problems with modularization [netty/netty/#7266](https://github.com/netty/netty/issues/7266) [netty/netty/#7268](https://github.com/netty/netty/pull/7268)
Unable to use Netty native transports in modularized Java 9 projects

Automatic module name for the epoll transport would be: `netty.transport.native.epoll`
But `native` is reserved keyword in java

**Comment by qweek** [comment]
Before merge, someone should check that `publish` step will be executed once

File: build.gradle [hunk 1]
```diff
@@ -76,6 +73,12 @@ subprojects {
         jmhVersion = '1.19'
     }
 
+    compileJava {
+        sourceCompatibility = jdkVersion
+        targetCompatibility = jdkVersion
```
Comment (yschimke) [inline build.gradle]: Shouldn't this stay 1.8 in all cases?

Comment (qweek) [reply build.gradle]: jdkVersion 1.9 used only for oraclejdk9 image on Travis
Why you want to use 1.8 in this case?

Comment (yschimke) [reply build.gradle]: I don't have any objections to compiling and running with JDK 9, that's a useful test.

But I don't think we want JDK 9 features used in source, or JDK9 specific jars.  Was this the aim?

Comment (qweek) [reply build.gradle]: Got it, no it's just compatibility test.

**Comment by yschimke** [comment]
If you want it as a compatibility test and that requires changing the source or target, then update the publish step as you suggest to only publish for 1.8 

https://github.com/rsocket/rsocket-java/blob/1.0.x/gradle/buildViaTravis.sh

**Comment by robertroeser** [comment]
why are we moving to java 9 before we release 1.0? I'm closing this.

**Comment by qweek** [comment]
We are not moving, just testing
Someone might want to use RSocket library in Java 9 project

**Pull request closed without merging**